src/Fabulous.AST/Widgets/ControlFlow/IfThen.fs

```diff
@@ -0,0 +1,51 @@
+namespace Fabulous.AST
+
+open System.Runtime.CompilerServices
+open FSharp.Compiler.Text
+open Fabulous.AST.StackAllocatedCollections
+open Fabulous.AST.StackAllocatedCollections.StackList
+open Fantomas.Core.SyntaxOak
+
+module IfThen =
+    let ThenExpr = Attributes.defineWidgetCollection "ThenExpr"
+    let IfExpr = Attributes.defineWidget "IfExpr"
+
+    let WidgetKey =
+        Widgets.register "IfThen" (fun widget ->
+            let ifExpr = Helpers.getNodeFromWidget<Expr> widget IfExpr
+
+            let thenExpr =
+                Helpers.getNodesFromWidgetCollection<ComputationExpressionStatement> widget ThenExpr
+
+            ExprIfThenNode(
+                IfKeywordNode.SingleWord(SingleTextNode("if", Range.Zero)),
+                ifExpr,
+                SingleTextNode("then", Range.Zero),
+                Expr.CompExprBody(ExprCompExprBodyNode(thenExpr, Range.Zero)),
+                Range.Zero
+            ))
+
+[<AutoOpen>]
+module IfThenBuilders =
+    type Fabulous.AST.Ast with
+
+        static member inline IfThen(exp: WidgetBuilder<Expr>) =
+            CollectionBuilder<ExprIfThenNode, ComputationExpressionStatement>(
```

Comment:
`ComputationExpressionStatement` is not correct in this case.
This is used when you have multiple statements following each other.
This should be `Expr` for the both the `conditionExpr` and the `thenExpr`.

Reply:
This was to able to support multiple Expr on the `IfThen`. But looks like from the AST POV not correct. So I will change this to simple WidgetBuilder